dotnet/src/IntegrationTests/Processes/ProcessTests.cs

```diff
@@ -0,0 +1,256 @@
+        Assert.NotNull(repeatStepState?.State);
+        Assert.Equal(string.Join(" ", Enumerable.Repeat(testInput, 4)), repeatStepState.State.LastMessage);
+    }
+
+    /// <summary>
+    /// Tests a process with three steps where the third step is a nested process. Events from the inner process
+    /// are routed to the outer process.
+    /// </summary>
+    /// <returns>A <see cref="Task"/></returns>
+    [Fact]
+    public async Task NestedProcessInnerToOuterDoesNotWorkWithInternalEventAsync()
+    {
+        // Arrange
+        OpenAIConfiguration configuration = this._configuration.GetSection("OpenAI").Get<OpenAIConfiguration>()!;
+        this._kernelBuilder.AddOpenAIChatCompletion(
+            modelId: configuration.ModelId!,
+            apiKey: configuration.ApiKey);
+
+        Kernel kernel = this._kernelBuilder.Build();
+
+        // Create the outer process
+        var processBuilder = this.CreateLinearProcess("Outer");
+
+        // Create the inner process and add it as a step to the outer process
+        var nestedProcessStep = processBuilder.AddStepFromProcess(this.CreateLinearProcess("Inner"));
+
+        // Add a new external event to start the outer process and handoff to the inner process directly
+        processBuilder.OnExternalEvent(ProcessTestsEvents.StartInnerProcess)
+            .SendEventTo(nestedProcessStep.GetTargetForExternalEvent(ProcessTestsEvents.StartProcess));
+
+        // Route the last step of the inner process to trigger the echo step of the outer process
+        nestedProcessStep.OnEvent(ProcessTestsEvents.OutputReadyInternal)
+            .SendEventTo(new ProcessFunctionTargetBuilder(processBuilder.Steps[0]));
+
+        // Build the outer process
+        var process = processBuilder.Build();
+
+        // Act
+        string testInput = "Test";
+        var processHandle = await process.StartAsync(kernel, new() { Id = ProcessTestsEvents.StartInnerProcess, Data = testInput });
+        var processInfo = await processHandle.GetStateAsync();
+
+        // Assert
+        var repeatStepState = processInfo.Steps.Where(s => s.State.Name == nameof(RepeatStep)).FirstOrDefault()?.State as KernelProcessStepState<StepState>;
+        Assert.NotNull(repeatStepState);
+        Assert.Null(repeatStepState.State?.LastMessage);
+    }
+
+    /// <summary>
+    /// Creates a simple linear process with two steps.
+    /// </summary>
+    private ProcessBuilder CreateLinearProcess(string name)
+    {
+        var processBuilder = new ProcessBuilder(name);
+        var echoStep = processBuilder.AddStepFromType<EchoStep>();
+        var repeatStep = processBuilder.AddStepFromType<RepeatStep>();
+
+        processBuilder.OnExternalEvent(ProcessTestsEvents.StartProcess)
+            .SendEventTo(new ProcessFunctionTargetBuilder(echoStep));
+
+        echoStep.OnFunctionResult(nameof(EchoStep.Echo))
```

Comment:
in case the Echo Step function had a name this should have been the FunctionName instead right?

Reply:
Yes, in this case `nameof(EchoStep.Echo)` is the name of the function.